### PR TITLE
allow banning of users

### PR DESCRIPTION
**Pull request by Rizziepit**

### CODE REVIEW

**Comment by Rizziepit** [comment]
@miltontony @smn ready for review. I moved `views/__init__.py` to `views/base.py`. You can ignore those changes.

File: unicore/comments/service/models.py [hunk 1]
```diff
@@ -50,6 +53,12 @@ def _pk_expression(cls, data):
     def pk_expression(self):
         return self.__class__._pk_expression(self.row_dict)
 
+    @classmethod
+    def match_all_expression(cls, **kwargs):
+        expression = [cls.__table__.c[k] == v
+                      for k, v in kwargs.iteritems()]
+        return and_(*expression)
+
     def to_dict(self):
```
Comment (smn) [inline unicore/comments/service/models.py:62]: Why not also implement `__dict__` ?

Comment (Rizziepit) [inline unicore/comments/service/models.py:62]: The semantics of `__dict__` and `to_dict` differ by design. `to_dict` returns a copy and it explicitly includes only fields that the model "exports" - by default only database columns.

I like using the standard `__dict__` implementation for introspection, but if there's a good use case for overriding it...

Comment (smn) [inline unicore/comments/service/models.py:62]: Excellent.

**Comment by smn** [comment]
:+1:

File: unicore/comments/service/views/comments.py [hunk 2]
```diff
@@ -28,6 +30,20 @@
...
+    cols = BannedUser.__table__.c
+    expression = cols.user_uuid == user_uuid
+    expression = and_(
+        expression,
+        or_(cols.app_uuid == app_uuid, cols.app_uuid.is_(None)))
+
+    query = BannedUser.__table__.select(exists().where(expression))
+    result = yield connection.execute(query)
+    result = yield result.scalar()
+    returnValue(bool(result))
```
Comment (smn) [inline unicore/comments/service/views/comments.py]: Sometimes these things are easier to write with just plain Deferreds:

``` python
d = connection.execute(query)
d.addCallback(lambda result: result.scalar())
d.addCallback(lambda count: bool(count))
return d
```

Comment (Rizziepit) [inline unicore/comments/service/views/comments.py]: I'll keep that in mind. Is there a notable difference in speed?

Comment (smn) [inline unicore/comments/service/views/comments.py]: Not significant enough for our use cases.

Comment (Rizziepit) [inline unicore/comments/service/views/comments.py]: I've updated `is_banned_user`

**Pull request merged**